internal/juju/client.go

```diff
@@ -104,6 +112,11 @@ func NewClient(ctx context.Context, config ControllerConfiguration) (*Client, er
 		defaultJAASCheck = true
 	}
 
+	user := config.Username
+	if config.ClientID != "" {
+		user = fmt.Sprintf("%s%s", config.ClientID, serviceAccountSuffix)
```

Comment:
A small thing but in the future we might want to force the user to add the "@serviceaccount" suffix themselves. So you could consider only adding the suffix if it's not already present. Lots of "ifs" so feel free to ignore this.

Reply:
done